### PR TITLE
ISPN-12232 CLI configuration properties

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-12232

### CODE REVIEW

File: cli/cli-client/src/main/java/org/infinispan/cli/impl/ContextImpl.java [hunk 1]
```diff
@@ -59,14 +82,41 @@ public boolean isConnected() {
...
+   }
+
+   @Override
+   public void saveProperties() {
+      Path configFile = configPath.resolve(CONFIG_FILE);
+      try {
+         Files.createDirectories(configPath);
+         Writer w = Files.newBufferedWriter(configFile);
+         properties.store(w, null);
+         Util.close(w);
```
Comment (ryanemerson) [inline cli/cli-client/src/main/java/org/infinispan/cli/impl/ContextImpl.java:114]: This should either be in a finally block or we should use a try-with-resources when creating the Writer.

**Review by ryanemerson — APPROVED** [review]
Overall LGTM, just one small comment. @oraNod can you review the docs please.

File: cli/cli-client/src/main/java/org/infinispan/cli/logging/Messages.java [hunk 2]
```diff
@@ -200,4 +200,13 @@
 
    @Message(value = "Attempted to use a different realm '%s' than the already existing one '%s'")
    IllegalArgumentException userToolWrongRealm(String realm1, String realm2);
+
+   @Message(value = "Unable to load CLI configuration from `%s`. Using defaults.")
```
Comment (oraNod) [inline cli/cli-client/src/main/java/org/infinispan/cli/logging/Messages.java:204]:
```suggestion
   @Message(value = "Cannot load CLI configuration from `%s`. Using default properties.")
```

File: cli/cli-client/src/main/java/org/infinispan/cli/logging/Messages.java [hunk 3]
```diff
@@ -200,4 +200,13 @@
...
+   String configLoadFailed(String path);
+
+   @Message(value = "Unable to store CLI configuration to '%s'.")
```
Comment (oraNod) [inline cli/cli-client/src/main/java/org/infinispan/cli/logging/Messages.java:207]:
```suggestion
   @Message(value = "Cannot store CLI configuration to '%s'.")
```

File: cli/cli-client/src/main/java/org/infinispan/cli/logging/Messages.java [hunk 4]
```diff
@@ -200,4 +200,13 @@
...
+   String configStoreFailed(String path);
+
+   @Message(value = "Wrong argument count: %d.")
```
Comment (oraNod) [inline cli/cli-client/src/main/java/org/infinispan/cli/logging/Messages.java:210]:
```suggestion
   @Message(value = "Incorrect argument count: %d.")
```

**Review by oraNod — APPROVED** [review]
LGTM

File: cli/cli-client/src/main/java/org/infinispan/cli/impl/ContextImpl.java (hunk 1 above)
Comment (tristantarrant) [reply cli/cli-client/src/main/java/org/infinispan/cli/impl/ContextImpl.java]: Done.

**Comment by tristantarrant** [comment]
I've squashed @oraNod 's changes and addressed @ryanemerson 's comment

**Comment by ryanemerson** [comment]
Thanks @tristantarrant

**Pull request merged**